assets/helpers/payPalContributionsCheckout/payPalContributionsCheckoutActions.js

```diff
@@ -33,53 +34,50 @@ export function payPalContributionsError(message: string): Action {
 }
 
 
-export function paypalContributionsRedirect(): Function {
+export function paypalContributionsRedirect(
+  amount: number,
+  intCmp: ?string,
+  isoCountry: IsoCountry,
+  errorHandler: (string) => void): void {
 
   const PAYPAL_CONTRIBUTION_ENDPOINT:string = window.guardian.contributionsPayPalEndpoint;
 
-  return (dispatch, getState: () => CombinedState) => {
-
-    const state = getState();
-
-    dispatch(payPalContributionsSubmitPayment());
-
-
-    const postData: PayPalPostData = {
-      countryGroup: 'uk',
-      amount: state.payPalContributionsCheckout.amount,
-      intCmp: state.intCmp,
-      supportRedirect: true,
-      /*
-       TODO: pass these argument to contributions in order to improve
-       the tracking of one-off contributions.
-
-       cmp: state.data.cmpCode,
-       refererPageviewId: state.data.refererPageviewId,
-       refererUrl: state.data.refererUrl,
-       ophanPageviewId: state.data.ophan.pageviewId,
-       ophanBrowserId: state.data.ophan.browserId,
-       ophanVisitId: state.data.ophan.visitId
-      */
-    };
-
-    const fetchOptions: Object = {
-      credentials: 'same-origin',
-      method: 'POST',
-      headers: {
-        Accept: 'application/json',
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(postData),
-    };
-
-    fetch(PAYPAL_CONTRIBUTION_ENDPOINT, fetchOptions)
-      .then((response) => {
-        if (response.ok) {
-          return response.json();
-        }
-        throw response;
-      })
-      .then((res) => { window.location = res.approvalUrl; })
-      .catch(() => dispatch(payPalContributionsError('Sorry, an error occurred, please try again or use another payment method.')));
+  const guCountry = toGuCountryCode(isoCountry);
+  const postData: PayPalPostData = {
+    countryGroup: guCountry,
+    amount,
+    intCmp,
+    supportRedirect: true,
+    /*
+     TODO: pass these argument to contributions in order to improve
+     the tracking of one-off contributions.
+
+     cmp: state.data.cmpCode,
+     refererPageviewId: state.data.refererPageviewId,
+     refererUrl: state.data.refererUrl,
+     ophanPageviewId: state.data.ophan.pageviewId,
+     ophanBrowserId: state.data.ophan.browserId,
+     ophanVisitId: state.data.ophan.visitId
+    */
   };
+
+  const fetchOptions: Object = {
+    credentials: 'same-origin',
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(postData),
+  };
+
+  fetch(PAYPAL_CONTRIBUTION_ENDPOINT, fetchOptions)
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw response;
+    })
+    .then((res) => { window.location = res.approvalUrl; })
+    .catch(() => errorHandler('Sorry, an error occurred, please try again or use another payment method.'));
 }
```

Comment:
@JustinPinner is this the file that is made redundant by `payPalContributionsCheckout.js`, but for which we're leaving the removal to a future refactor PR, in order to avoid reworking multiple reducers?